### PR TITLE
feat(WebexComponents): add "My preview" banner 

### PR DESCRIPTION
**Pull request by patricia0817**
Add a "My preview" banner to the local media component, in the interstitial meeting. The banner should only be present on the "big" video, and only when the local video is available. 
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-254211

### CODE REVIEW

File: src/themes/dark.scss [hunk 1]
```diff
@@ -1,3 +1,6 @@
 .wxc-theme-dark {
   --wxc-meeting-background: linear-gradient(180deg, #1A1A1A 0%, #0F0F0F 100%);
+  // --wxc-banner-background: linear-gradient(177.75deg, rgba(0, 0, 0, 0) 1.9%, rgba(0, 0, 0, 0.3) 50.23%, rgba(0, 0, 0, 0.8) 98.11%);
+  --wxc-banner-background: linear-gradient(3.25deg, rgba(0, 0, 0, 0) 1.9%, rgba(0, 0, 0, 0.3) 50.23%, rgba(0, 0, 0, 0.8) 98.11%);
```
Comment (patricia0817) [inline src/themes/dark.scss:4]: should be 2.25

File: src/components/generic/Banner/Banner.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import webexComponentClasses from '../../helpers';
+
+/**
+ * Banner component.
+ *
+ * @param {object} props  Data passed to the component
+ * @param {React.ReactNode[]} props.children  List of children
```
Comment (cipak) [inline src/components/generic/Banner/Banner.jsx:9]: i think `JSX.Element` is a better choice for type (can be text, one node, array of nodes)

File: src/components/generic/Banner/Banner.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner {
+  background: var(--wxc-banner-background);
+  height: 120px;
```
Comment (cipak) [inline src/components/generic/Banner/Banner.scss:3]: the height should be auto, put some paddings to aquire the total height in figma

File: src/components/generic/Banner/Banner.scss [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+  .content {
```
Comment (cipak) [inline src/components/generic/Banner/Banner.scss:5]: i guess this class is no longer used, move the css props to the parent

File: src/components/generic/Banner/Banner.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    color: var(--wxc-banner-content-color);
+    font-family: CiscoSans;
+    line-height: 150%;
```
Comment (cipak) [inline src/components/generic/Banner/Banner.scss:8]: no need for that, use paddings instead

File: src/components/generic/Banner/Banner.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    font-size: 16px;
+    text-align: center;
```
Comment (cipak) [inline src/components/generic/Banner/Banner.scss:10]: or display: flex, justify-content: center on parent

File: src/themes/dark.scss [hunk 7]
```diff
@@ -1,3 +1,5 @@
 .wxc-theme-dark {
   --wxc-meeting-background: linear-gradient(180deg, #1A1A1A 0%, #0F0F0F 100%);
+  --wxc-banner-background: linear-gradient(2.25deg, rgba(0, 0, 0, 0) 1.9%, rgba(0, 0, 0, 0.3) 50.23%, rgba(0, 0, 0, 0.8) 98.11%);
+  --wxc-banner-content-color: rgba(255, 255, 255, 0.95);
```
Comment (cipak) [inline src/themes/dark.scss:4]: --wxc-banner-text-color

File: src/components/WebexLocalMedia/WebexLocalMedia.jsx [hunk 8]
```diff
@@ -58,8 +59,22 @@ export default function WebexLocalMedia({
...
-        /* eslint-disable-next-line jsx-a11y/media-has-caption */
-        stream ? <video ref={ref} playsInline autoPlay /> : disabledVideo
+        stream
+          ? (
+            <>
+              {
+                mediaType === 'video'
+                  && (
+                    <Banner className="my-preview">
+                      <div className="content">My Preview</div>
```
Comment (cipak) [inline src/components/WebexLocalMedia/WebexLocalMedia.jsx:69]: no need for an additional div here, just:
```js
<Banner ...>My Preview</Banner>

File: src/components/WebexLocalMedia/WebexLocalMedia.jsx [hunk 9]
```diff
@@ -58,8 +59,22 @@ export default function WebexLocalMedia({
   return (
     <div ref={mediaRef} className={cssClasses} style={style}>
       {
-        /* eslint-disable-next-line jsx-a11y/media-has-caption */
-        stream ? <video ref={ref} playsInline autoPlay /> : disabledVideo
+        stream
```
Comment (cipak) [inline src/components/WebexLocalMedia/WebexLocalMedia.jsx:62]: this looks way too expanded:)
try this:
```js
let content;
if (stream) {
  content = <>
    {mediaType === 'video' && <Banner>...</Banner>}
    <video..>
  <>;
}
else {
  content = ID ? <WebexAvatar personID={ID} displayStatus={false} /> : <Spinner />;
}

File: src/components/WebexLocalMedia/WebexLocalMedia.jsx [hunk 10]
```diff
@@ -58,8 +59,22 @@ export default function WebexLocalMedia({
...
+            <>
+              {
+                mediaType === 'video'
+                  && (
+                    <Banner className="my-preview">
+                      <div className="content">My Preview</div>
+                    </Banner>
+                  )
+              }
+              {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
```
Comment (cipak) [inline src/components/WebexLocalMedia/WebexLocalMedia.jsx:73]: what happens if we remove this eslint directive?

Comment (cipak) [reply src/components/WebexLocalMedia/WebexLocalMedia.jsx:73]: ok, seems to be an accessibility issue, you can leave it as it is https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/media-has-caption.md

File: src/components/WebexLocalMedia/WebexLocalMedia.jsx [hunk 11]
```diff
@@ -11,6 +11,7 @@ import {
   useMeeting,
   useStream,
 } from '../hooks';
+import Banner from '../generic/Banner/Banner';
```
Comment (cipak) [inline src/components/WebexLocalMedia/WebexLocalMedia.jsx:14]: the generic components should be above the more specific components (eg, above WebexAvatar)

File: src/components/WebexLocalMedia/WebexLocalMedia.scss [hunk 12]
```diff
@@ -10,6 +10,12 @@
 
   background: black;
 
+  .my-preview {
+    position: absolute;
+    top: 25%;
```
Comment (cipak) [inline src/components/WebexLocalMedia/WebexLocalMedia.scss:15]: isn't the banner supposed to be displayed at the top of the local media component?

File: src/components/generic/Banner/Banner.jsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Banner component.
+ *
+ * @param {object} props  Data passed to the component
+ * @param {JSX.Element} props.children  List of children
+ * @param {string} props.className  Custom CSS class to apply
+ * @returns {object} JSX of the component
+ */
+export default function Banner({children, className}) {
```
Comment (cipak) [inline src/components/generic/Banner/Banner.jsx:13]: also, `style`

File: src/components/generic/Banner/Banner.scss [hunk 14]
```diff
@@ -0,0 +1,11 @@
+.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner {
+  display: flex;
+  justify-content: center;
+  background: var(--wxc-banner-background);
+  color: var(--wxc-banner-text-color);
+  font-family: CiscoSans;
+  font-size: 16px;
+  height: 120px;
+  padding: 12px 0 84px 0;
```
Comment (cipak) [inline src/components/generic/Banner/Banner.scss:9]: I don't think height and vertical paddings are both necessary, please leave only one of them

File: src/components/generic/_generic.scss [hunk 15]
```diff
@@ -1 +1,2 @@
 @import 'Modal/Modal';
+@import 'Banner/Banner';
```
Comment (cipak) [inline src/components/generic/_generic.scss:2]: alphabetical

File: src/themes/legacy.scss [hunk 16]
```diff
@@ -1,3 +1,5 @@
 :root {
   --wxc-meeting-background: transparent;
+  --wxc-banner-background: transparent;
```
Comment (cipak) [inline src/themes/legacy.scss:3]: let's hide this banner with a different technique:

```css
:root {
  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner.my-preview { display: none; }
}
```
and in the theme file, eg dark.scss
```js
:root .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-theme-dark {
  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner.my-preview { display: block; }
}
```
we will remove these rules once the theming is complete.

File: src/themes/dark.scss [hunk 17]
```diff
@@ -1,6 +1,13 @@
 .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-theme-dark {
   --wxc-meeting-background: #{$wxc-gradation-webex-dark-primary};
   --wxc-text-color: #{$wxc-white-alpha-100};
+
+  //Banner
```
Comment (cipak) [inline src/themes/dark.scss:5]: no need for the comment

File: src/themes/dark.scss [hunk 18]
```diff
@@ -1,6 +1,13 @@
...
+  --wxc-banner-background: linear-gradient(2.25deg, rgba(0, 0, 0, 0) 1.9%, rgba(0, 0, 0, 0.3) 50.23%, rgba(0, 0, 0, 0.8) 98.11%);
```
Comment (cipak) [inline src/themes/dark.scss:6]: this gradient should already be defined in _colors.scss

File: src/themes/dark.scss [hunk 19]
```diff
@@ -15,10 +22,6 @@
 
   .wxc-old-icon { display: none; }
   .wxc-icon { display: block; }
-}
 
-.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-theme-dark {
-  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title {
-    display: block;
-  }
+  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title { display: block; }
```
Comment (cipak) [inline src/themes/dark.scss:26]: let's keep the separate block and put all temporary rules in that block (variables in one block, temporary rules in another block)

File: src/themes/legacy.scss [hunk 20]
```diff
@@ -1,10 +1,10 @@
...
   --wxc-text-color: #{$wxc-black-alpha-100};
+  --wxc-meeting-text-color: #000000;
 }
 
-.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title {
-  display: none;
-}
+.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title { display: none; }
+.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner.my-preview { display: none; }
 .wxc-old-icon { display: block; }
```
Comment (cipak) [inline src/themes/legacy.scss:9]: please prefix these, too

Comment (cipak) [reply src/themes/legacy.scss:9]: and remove the empty line above

File: src/themes/light.scss [hunk 21]
```diff
@@ -1,6 +1,13 @@
 .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-theme-light {
   --wxc-meeting-background: #{$wxc-gradation-webex-light-primary};
   --wxc-text-color: #{$wxc-black-alpha-100};
+
+  //Banner
+  --wxc-banner-background: linear-gradient(2.25deg, rgba(255, 255, 255, 0) 1.9%, rgba(255, 255, 255, 0.3) 50.23%, rgba(255, 255, 255, 0.8) 98.11%);
+  --wxc-banner-text-color: rgba(0, 0, 0, 0.95);
```
Comment (cipak) [inline src/themes/light.scss:7]: use color sass variables, they should already be available in _colors.scss

File: src/styles/_variables.scss [hunk 22]
```diff
@@ -22,7 +22,7 @@ $md-gray-95: #1C1C1C !default;
 $md-gray-100: #121212 !default;
 $md-yellow-40: #FC9D03 !default;
 
-$md-red-60: #D4371C !default; 
+$md-red-60: #D4371C !default;
```
Comment (cipak) [inline src/styles/_variables.scss:25]: if you didn't do any other modification to this file, let's leave it like it was, these colors will be all removed anyway

File: src/themes/dark.scss (hunk 18 above)
Comment (patricia0817) [reply src/themes/dark.scss:6]: it's not, but I will add it

File: src/components/generic/Banner/Banner.scss [hunk 23]
```diff
@@ -0,0 +1,8 @@
+.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner {
+  justify-content: center;
+  background: var(--wxc-banner-background);
+  color: var(--wxc-banner-text-color);
+  font-family: $brand-font-regular;
+  font-size: 16px;
```
Comment (cipak) [inline src/components/generic/Banner/Banner.scss:6]: use rem instead of px

File: src/styles/_variables.scss (hunk 22 above)
Comment (cipak) [reply src/styles/_variables.scss:25]: not resolved

File: src/themes/dark.scss [hunk 24]
```diff
@@ -3,17 +3,29 @@
...
+  --wxc-banner-background: #{$wxc-gradation-dark-transparent};
+  --wxc-banner-text-color: rgba(255, 255, 255, 0.95);
+
+  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner.my-preview { display: flex; }
+
+  // Icons
   --wxc-icon--color: #{$wxc-white-alpha-100};
   --wxc-icon--active-color: #{$wxc-green-50};
   --wxc-icon--muted-color: #{$wxc-red-50};
+  --wxc-icon--audio-microphone--active-color: #{$wxc-green-50};
```
Comment (cipak) [inline src/themes/dark.scss:16]: are these icon specific colors necessary?

File: src/themes/light.scss (hunk 21 above)
Comment (cipak) [reply src/themes/light.scss:8]: not resolved

Comment (cipak) [reply src/themes/light.scss:8]: and maybe it would be ok to use the standard wxc-text-color

File: src/themes/light.scss [hunk 25]
```diff
@@ -3,17 +3,29 @@
...
 
   .wxc-old-icon { display: none; }
   .wxc-icon { display: block; }
-}
 
-.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-theme-light {
-  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title {
-    display: block;
-  }
+  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title { display: block; }
```
Comment (cipak) [inline src/themes/light.scss:30]: it's ok to leave like it was, a separate block for these temporary logic properties:
```css
.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-theme-light {
  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member-roster .title { display: block; }
}

File: src/components/generic/Banner/Banner.scss (hunk 14 above)
Comment (cipak) [reply src/components/generic/Banner/Banner.scss:7]: please also use rem in this line

File: src/styles/_variables.scss (hunk 22 above)
Comment (cipak) [reply src/styles/_variables.scss:25]: still not resolved. Use another editor if possible. You have to add a blank space after the semicolon to revert this change.

File: src/themes/dark.scss [hunk 26]
```diff
@@ -3,7 +3,13 @@
   --wxc-text-color: #{$wxc-white-alpha-100};
   --wxc-meeting-border: 1px solid #{$wxc-white-alpha-30};
   --wxc-meeting-box-shadow: none;
-  
+
+  --wxc-banner-background: #{$wxc-gradation-dark-transparent};
+  --wxc-banner-text-color: rgba(255, 255, 255, 0.95);
```
Comment (cipak) [inline src/themes/dark.scss:8]: use color code

File: src/styles/_variables.scss (hunk 22 above)
Comment (cipak) [reply src/styles/_variables.scss:25]: still not resolved

File: src/themes/legacy.scss [hunk 27]
```diff
@@ -3,10 +3,10 @@
   --wxc-text-color: #{$wxc-black-alpha-100};
   --wxc-meeting-border: none;
   --wxc-meeting-box-shadow: none;
+  --wxc-meeting-text-color: #000000;
```
Comment (cipak) [inline src/themes/legacy.scss:6]: I think we can use the default `--wxc-text-color` var

File: src/components/generic/Banner/Banner.scss (hunk 14 above)
Comment (cipak) [reply src/components/generic/Banner/Banner.scss:7]: rem:)

File: src/themes/dark.scss [hunk 28]
```diff
@@ -3,7 +3,13 @@
   --wxc-text-color: #{$wxc-white-alpha-95};
   --wxc-meeting-border: 1px solid #{$wxc-white-alpha-30};
   --wxc-meeting-box-shadow: none;
-  
+
+  --wxc-banner-background: #{$wxc-gradation-dark-transparent};
+  --wxc-banner-text-color: #{$wxc-white-alpha-95};
+
+  .#{$WEBEX_COMPONENTS_CLASS_PREFIX}-banner.my-preview { display: flex; }
```
Comment (cipak) [inline src/themes/dark.scss:10]: this rule should go in the block of temporary logic below

File: src/themes/dark.scss [hunk 29]
```diff
@@ -3,7 +3,11 @@
   --wxc-text-color: #{$wxc-white-alpha-95};
   --wxc-meeting-border: 1px solid #{$wxc-white-alpha-30};
   --wxc-meeting-box-shadow: none;
-  
+
+  --wxc-banner-background: #{$wxc-gradation-dark-transparent};
+  --wxc-banner-text-color: #{$wxc-white-alpha-95};
+
+  // Icons
```
Comment (cipak) [inline src/themes/dark.scss:10]: this comment was removed on master, please remove it from your pr as well, thanks

**Review by cipak — APPROVED** [review]

**Pull request merged**